usersession/userd/settings.go

```diff
@@ -151,75 +279,73 @@ func (s *Settings) Get(setting string, sender dbus.Sender) (string, *dbus.Error)
 	}
 
 	cmd := exec.Command("xdg-settings", "get", setting)
-	output, err := cmd.CombinedOutput()
+	output, err := checkOutput(cmd, "get", setting)
 	if err != nil {
-		return "", dbus.MakeFailedError(fmt.Errorf("cannot get setting %s: %s", setting, osutil.OutputErr(output, err)))
+		return "", err
 	}
 
-	// avoid information leak: see https://github.com/snapcore/snapd/pull/4073#discussion_r146682758
-	snap, err := snapFromSender(s.conn, sender)
-	if err != nil {
-		return "", dbus.MakeFailedError(err)
+	return desktopFileFromOutput(s, output, sender)
+}
+
+// GetSub implements the 'GetSub' method of the 'io.snapcraft.Settings'
+// DBus interface.
+//
+// Example usage: dbus-send --session --dest=io.snapcraft.Settings --type=method_call --print-reply /io/snapcraft/Settings io.snapcraft.Settings.GetSub string:'default-url-scheme-handler' string:'irc'
+func (s *Settings) GetSub(setting string, subproperty string, sender dbus.Sender) (string, *dbus.Error) {
+	settingWithSub := settingWithSub(setting, subproperty)
```

Comment:
This looks wrong to me because the LHS var is the same name as the RHS function.

Reply:
Addressed in a0c708df0